sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java

```diff
@@ -621,7 +621,7 @@ private void validateAuthorizationClientProperties(String registrationId,
             }
 
             if (AZURE_CLIENT_REGISTRATION_ID.equals(registrationId)
-                && !AUTHORIZATION_CODE.equals(properties.getAuthorizationGrantType())) {
```

Comment:
refer: https://sonarcloud.io/project/issues?directories=sdk%2Fspring%2Fspring-cloud-azure-autoconfigure%2Fsrc%2Fmain%2Fjava%2Fcom%2Fazure%2Fspring%2Fcloud%2Fautoconfigure%2Faad%2Fproperties&resolved=false&types=CODE_SMELL&id=stliu_azure-sdk-for-java&open=AYALwwJVEEOpsvmOn-GD